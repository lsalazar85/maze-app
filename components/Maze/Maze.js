import {useEffect} from "react";

import useMaze from "../../hooks/useMaze";
import {MAZE_MATRIX as grid, STATUS} from "../../constants";
import {fetchingData} from "../../actions";

import Avatar from "@/components/Avatar";
import {
    MazeWrapper,
    MazeContainer,
    Row,
    Cube,
    Moves,
    MovesCount,
} from "./styles";

const Maze = () => {
    const { playerPosition } = useMaze()

    useEffect(() => {
        if(playerPosition.x === 11 && playerPosition.y === 10){
            try{
                fetchingData()
                alert(STATUS.message)
            } catch (e){
                console.log(e)
            }
        }
    }, [playerPosition])

    return(
        <MazeWrapper>
            <Moves><MovesCount>moves: {playerPosition.moves}</MovesCount></Moves>
            <MazeContainer>
                <Avatar position={playerPosition} />
                {grid.map((subArray, idx) => (
                    <Row key={idx}>
                        {subArray.map((subItem, idx) => (
                            <Cube key={idx} bg={subItem} />
                        ))}
                    </Row>
                ))}
            </MazeContainer>
        </MazeWrapper>
    )
}

export default Maze