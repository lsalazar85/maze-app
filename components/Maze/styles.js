import styled from "styled-components";

export const MazeContainer = styled.div`
  width: 40rem;
  height: 40rem;
  border: 1px solid red;
  position: relative;
`