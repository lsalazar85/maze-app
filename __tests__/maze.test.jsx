/**
 * @jest-environment jsdom
 */

import  { render } from '@testing-library/react'
import Maze from "@/components/Maze";

describe('Maze', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<Maze />)

        expect(asFragment()).toBeDefined();
        expect(asFragment()).toMatchSnapshot();
    })
})