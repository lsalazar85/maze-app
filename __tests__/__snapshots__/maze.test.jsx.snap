// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maze should render correctly 1`] = `
<DocumentFragment>
  <main
    class="styles__MazeWrapper-sc-1w8rlqt-0 cVvARo"
  >
    <div
      class="styles__Moves-sc-1w8rlqt-4 evCIVo"
    >
      <span
        class="styles__MovesCount-sc-1w8rlqt-5 gBCDKc"
      >
        moves: 0
      </span>
    </div>
    <section
      class="styles__MazeContainer-sc-1w8rlqt-1 dxngsq"
    >
      <div
        class="styles__AvatarContainer-sc-djljgk-0 fhuxRF"
        x="1"
        y="0"
      >
        <div
          class="styles__AvatarIcon-sc-djljgk-1 lhygVp"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 eecYHy"
        />
      </div>
      <div
        class="styles__Row-sc-1w8rlqt-2 bcEcFj"
      >
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
        <div
          class="styles__Cube-sc-1w8rlqt-3 kjSNVW"
        />
      </div>
    </section>
  </main>
</DocumentFragment>
`;
